src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java

```diff
@@ -922,12 +949,12 @@ private void createSession(final HttpServletRequest request, final HttpServletRe
             sessionInformationBuilder.put(SESSION_TOKEN, userSessionToken);
             sessionInformationBuilder.put(DATE_EXPIRES, sessionExpiryDate);
 
-            if (partialLoginFlag) {
-                partialLoginFlagString = String.valueOf(true);
-                sessionInformationBuilder.put(PARTIAL_LOGIN_FLAG, partialLoginFlagString);
+            for (AuthenticationCaveat c : authenticationCaveats) {
+                caveatFlags.add(c.toString());
```

Comment:
`authenticationCaveats` is a Set, which you then iterate over to create the list `caveatFlags`. There's no guarantee of ordering here, which matters when this is passed into `calculateSessionHMAC` which then uses the list in order and will fail to recognise two identical sets of caveats if the Set happens to order them differently sometimes.

Whilst at present there is at most one caveat, if we're going to write it in a way that accepts more than one caveat, we might as well future-proof it. Something like:
```java
caveatFlags = authenticationCaveats.stream().map(AuthenticationCaveat::toString).sorted().collect(Collectors.toList());
```
would at least impose a consistent order. Better, perhaps, would be to make `calculateSessionHMAC` take a Set and do the ordering itself so we don't have to worry about all places that use it?

Reply:
>Better, perhaps, would be to make calculateSessionHMAC take a Set and do the ordering itself so we don't have to worry about all places that use it?
Yep, I think that makes the most sense.